scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala

```diff
@@ -10,87 +10,83 @@ import scala.meta.tokens.Token.LeftBracket
 
 class SoftKeywords(dialect: Dialect) {
 
-  private def matches(token: Token, name: String, isEnabled: => Boolean): Boolean = {
-    isEnabled && token.toString() == name
-  }
-
   @classifier
   trait KwAs {
     val name = "as"
-    def unapply(token: Token): Boolean = {
-      matches(token, name, dialect.allowAsForImportRename)
-    }
+    @inline def isEnabled = dialect.allowAsForImportRename
+    @inline final def unapply(token: Token): Boolean = isEnabled && name == token.toString
+    @inline final def unapply(token: String): Boolean = isEnabled && name == token
```

Comment:
@kitbellew could you specify why this is needed? Are you going to use `unapply(token: String)`  in scalafmt?

Reply:
most immediately, in scalameta, in the next commit. to replace `token.is[soft.A] || token.is[soft.B] || ...`, which invokes `token.toString` multiple times.

Reply:
Maybe let's included it in this PR?

Reply:
@tgodzik can't include it yet, that change also depends on a commit which _itself_ depends on #2628.

Reply:
ok, let me refactor that part out and include it here, give me a minute

Reply:
done, see other comment